sdk/go/common/workspace/plugins.go

```diff
@@ -1342,11 +1342,9 @@ func getPluginInfoAndPath(
 		for j, p2 := range projectPlugins {
 			if j < i {
 				if p2.Kind == p1.Kind && p2.Name == p1.Name {
-					if p1.Version != nil && p2.Version != nil && p2.Version.Equals(*p1.Version) {
-						return nil, "", fmt.Errorf(
-							"multiple project plugins with kind %s, name %s, version %s",
-							p1.Kind, p1.Name, p1.Version)
-					}
+					return nil, "", fmt.Errorf(
```

Comment:
Not allowing aws@v4 and aws@v5 anymore? Per version stuff seemed to be working ok before.

Reply:
Maybe just match major versions?. When I was testing, the exact version matching was fairly brittle and would often cause the plugin to be missed.

Reply:
That can be a separate PR though

Reply:
This looks like it would affect Pulumi YAML and plugin resolution in general.

Reply:
This specific check only looks at linked plugins - it was introduced in #10146